### ISSUE TITLE
NVDA fails to interact with some web controls on Google pages if using UIA for Microsoft Edge

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#14612]
### Steps to reproduce:
1. From NVDA's advanced settings, set the option "Use UIA with Microsoft Edge and other Chromium based browsers when available" to "Yes".
2. Open Microsoft Edge.
3. Go to youtube.com.
4. Go to the toggle button "Guide" and press ENTER to activate it.

### Actual behavior:
Nothing happens.

### Expected behavior:
The guide panel on the left should appear/disappear, depending on its status and NVDA should report "checked"/"Not checked" as a status for the focused "Guide" toggle button.

### NVDA logs, crash dumps and other attachments:
[NVDA_MSEdge_UIA.log](https://github.com/nvaccess/nvda/files/10664718/NVDA_MSEdge_UIA.log)

#### NVDA installed/portable/running from source:
Installed.

#### NVDA version:
NVDA version: alpha-27627,79ad4f07.

#### Windows version:
Operating system: Windows 11 Pro version 22H2 (build 22621.1105), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge version: 109.0.1518.78, in bulgarian.

#### Does the issue still occur after restarting your computer?
Yes, it does.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - NVDA 2022.4 and the result is the same.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes, it is.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
I haven't done that recently, but previously I have for some other diagnostics and the problem was occurring before that and after it, so this is not relevant.

**Comment by k-kolev1985** [comment]
Actually, it seems to affect controls on other web pages as well (e.g. the button to submit the new issue here on GitHub).

**Comment by michaelDCurran** [comment]
In this scenario, the browse mode caret is positioned on a graphic in a grouping in the guide toggle button.
NvDA tries to invoke the graphic but nothing happens.
In UIA browseMode's _activateNVDAObject or somewhere higher, NvDA should either walk up ancestors until doAction succeeds.
Or perhaps in activatePosition, care should be taken to use the deepest NVDAObject within the presented content. In this case it is the guide button, not the graphic or grouping.

**Comment by Adriani90** [comment]
Here is also a minimal test case for a link that should open the github website from #14718..

```
<!DOCTYPE HTML>
<html>
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
    <title>Test Demo</title>
</head>
<body>
    <a itemtype="url" href="https://github.com/">open  Github <span style='color:red'>[website]</span></a>
</body>
</html>
```

**Comment by cary-rowen** [comment]
Following the steps of @k-kolev1985, I wrote another minimal demo.
```html
<!DOCTYPE html>
<html>
<head>
	<title>Button Demo</title>
</head>
<body>
	<button id="button" aria-label="Guide" aria-pressed="false">
    </button>
	<script>
		const button = document.getElementById("button");
		button.addEventListener("click", function() {
			const isPressed = button.getAttribute("aria-pressed") === "true";
			button.setAttribute("aria-pressed", !isPressed);
		});
	</script>
</body>
</html>
```
